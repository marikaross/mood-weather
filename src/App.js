import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <a
            className="App-link"
            href="https://darksky.net/poweredby/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Powered by Dark Sky
          </a>
        </header>
      </div>
    );
  }
}

export default App;
