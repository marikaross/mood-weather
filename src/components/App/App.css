.medium-app {
  text-align: center;
  background: linear-gradient(rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.25)), url("../../assets/nice.jpg") no-repeat;
  background-size: cover;
  height: 100vh;
}

.hot-app {
  text-align: center;
  background: linear-gradient(rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.25)), url("../../assets/palm-trees.jpg") no-repeat;
  background-size: cover;
  height: 100vh;
}

.cold-app {
  text-align: center;
  background: linear-gradient(rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.25)), url("../../assets/cold.jpg") no-repeat;
  background-size: cover;
  height: 100vh;
}

h1 {
  color: #f5f5f5;
  margin: 0;
}

.App-link {
  padding: 10px;
  color: #f5f5f5;
  font-size: calc(8px + 2vmin);
}


