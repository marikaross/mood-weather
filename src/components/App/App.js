import React, { Component } from 'react';
import Search from '../Search/Search.js';
import apiKey from '../../apiKey.js'; 
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      location: {}
    }
  }

  componentDidMount() {

  }

  setLocation = (coordinates) => {
    this.setState({location: coordinates})
  }

  getWeather = async () => {
    const allInfo = await fetch(`https://api.darksky.net/forecast/${apiKey}/${this.state.location.userLat},${this.state.location.userLong}`)
    const response = await allInfo.json()

  }

  render() {
    return (
      <div className="App">
        <Search setLocation={this.setLocation}/>
        <header className="App-header">
          <a
            className="App-link"
            href="https://darksky.net/poweredby/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Powered by Dark Sky
          </a>
        </header>
      </div>
    );
  }
}

export default App;
