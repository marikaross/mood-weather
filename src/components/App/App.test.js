import React from 'react';
import App from './App.js';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';


describe('App', () => {
  let wrapper, 
  mockSetLocation,
  mockSnakeCase,
  mockSetConditions,
  mockGetWeather,
  mockLocation,
  mockConditons;

  beforeEach(() => {
    mockSetLocation = jest.fn();
    mockSnakeCase = jest.fn();
    mockSetConditions = jest.fn();
    mockGetWeather = jest.fn();
    mockLocation = {userLat: 40.016457, userLong: -105.285884}
    mockConditons = {icon: 'FOG', temperature: 1000, summary: 'Hell'}

    wrapper = shallow(<App 
      setLocation={mockSetLocation}
      setconditions={mockSetConditions}
      snakeCase={mockSnakeCase}
      getWeather={mockGetWeather}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state with location and conditions', () => {
    let expected = {
      location: {},
      conditions: {}
    }

    expect(wrapper.state('location')).toEqual(expected.location)
    expect(wrapper.state('conditions')).toEqual(expected.conditions)
  })

  it('should change the location in state based on user input', () => {
    const expected = {conditions: {}, location: {userLat: 40.016457, userLong: -105.285884}}
    wrapper.checkLocation = jest.fn()
    wrapper.instance().setLocation(mockLocation)
    expect(wrapper.state()).toEqual(expected)
  })

  it('should call getWeather with the appropriate url', () => {
     window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({
        
      }),
  })


})
