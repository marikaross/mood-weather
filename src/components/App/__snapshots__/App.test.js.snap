// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    getWeather={[MockFunction]}
    setLocation={[MockFunction]}
    setconditions={[MockFunction]}
    snakeCase={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Find Your Weather
        </h1>,
        <div
          className="search-display"
        >
          <Search
            setLocalStorage={[Function]}
            setLocation={[Function]}
          />
          <Display
            conditions={Object {}}
          />
          <a
            className="App-link"
            href="https://darksky.net/poweredby/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Powered by Dark Sky
          </a>
        </div>,
      ],
      "className": "medium-app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find Your Weather",
          "className": "title",
        },
        "ref": null,
        "rendered": "Find Your Weather",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Search
              setLocalStorage={[Function]}
              setLocation={[Function]}
            />,
            <Display
              conditions={Object {}}
            />,
            <a
              className="App-link"
              href="https://darksky.net/poweredby/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Powered by Dark Sky
            </a>,
          ],
          "className": "search-display",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "setLocalStorage": [Function],
              "setLocation": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "conditions": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Powered by Dark Sky",
              "className": "App-link",
              "href": "https://darksky.net/poweredby/",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Powered by Dark Sky",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Find Your Weather
          </h1>,
          <div
            className="search-display"
          >
            <Search
              setLocalStorage={[Function]}
              setLocation={[Function]}
            />
            <Display
              conditions={Object {}}
            />
            <a
              className="App-link"
              href="https://darksky.net/poweredby/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Powered by Dark Sky
            </a>
          </div>,
        ],
        "className": "medium-app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find Your Weather",
            "className": "title",
          },
          "ref": null,
          "rendered": "Find Your Weather",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Search
                setLocalStorage={[Function]}
                setLocation={[Function]}
              />,
              <Display
                conditions={Object {}}
              />,
              <a
                className="App-link"
                href="https://darksky.net/poweredby/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Powered by Dark Sky
              </a>,
            ],
            "className": "search-display",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "setLocalStorage": [Function],
                "setLocation": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "conditions": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Powered by Dark Sky",
                "className": "App-link",
                "href": "https://darksky.net/poweredby/",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Powered by Dark Sky",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
